conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -43,6 +44,20 @@ def __init__(self, conanfile, namespace=None):
         self.fpic = self._conanfile.options.get_safe("fPIC")
         self.msvc_runtime_flag = self._get_msvc_runtime_flag()
 
+        # standard build toolchains env vars
+        self.cc = self._cc
+        self.cxx = self._cxx
+        self.ld = self._ld
+        self.ar = self._ar
+        self.nm = self._nm
+        self.objdump = self._objdump
+        self.ranlib = self._ranlib
+        self.strip = self._strip
+
+        # Wrappers paths for tools not compatible with autotools, like msvc
+        self.compile_wrapper = None
+        self.arlib_wrapper = None
```

Comment:
You need to move these forwards, before `self.cc` etc are assigned.

Reply:
moved to constructor arguments: https://github.com/conan-io/conan/commit/40fdb2000fd727a4f2f1309c3fb848910b2cb0d8